build.gradle

```diff
@@ -566,6 +566,39 @@ task e2eTests {
     e2eTests.dependsOn "e2eTestTry${id}"
 }
 
+task e2eUnstableTests {
+    description "Runs the unstable E2E test suite and retries failed test up to ${numOfTestRetries} times."
+    group "Test"
+}
+
+(1..numOfTestRetries + 1).each { id ->
+    def isFirstTry = id == 1
+    def isLastRetry = id == numOfTestRetries + 1
+
+    task "e2eUnstableTestTry${id}"(type: Test) {
+        useTestNG()
+        options.suites isFirstTry ? "src/e2e/resources/testng-unstable-e2e.xml" : file("${buildDir}/reports/e2e-unstable-test-try-${id - 1}/testng-failed.xml")
```

Comment:
Similarly here, instead of duplicating the entire thing, consider using `-D` or `-P` flag to differentiate the XML file being used